nip47/notifications/nip47_notifier.go

```diff
@@ -139,39 +142,65 @@ func (notifier *Nip47Notifier) notifySubscriber(ctx context.Context, app *db.App
 		}).WithError(err).Error("Failed to stringify notification")
 		return
 	}
-	msg, err := nip04.Encrypt(string(payloadBytes), ss)
-	if err != nil {
-		logger.Logger.WithFields(logrus.Fields{
-			"notification": notification,
-			"appId":        app.ID,
-		}).WithError(err).Error("Failed to encrypt notification payload")
-		return
+
+	var msg string
+	if legacy {
+		ss, err := nip04.ComputeSharedSecret(app.AppPubkey, appWalletPrivKey)
+		if err != nil {
+			logger.Logger.WithFields(logrus.Fields{
+				"notification": notification,
+				"appId":        app.ID,
+			}).WithError(err).Error("Failed to compute shared secret")
+			return
+		}
+		msg, err = nip04.Encrypt(string(payloadBytes), ss)
+		if err != nil {
+			logger.Logger.WithFields(logrus.Fields{
+				"notification": notification,
+				"appId":        app.ID,
+			}).WithError(err).Error("Failed to encrypt notification payload")
+			return
+		}
+	} else {
+		ck, err := nip44.GenerateConversationKey(app.AppPubkey, appWalletPrivKey)
+		if err != nil {
+			logger.Logger.WithFields(logrus.Fields{
+				"notification": notification,
+				"appId":        app.ID,
+			}).WithError(err).Error("Failed to compute conversation key")
+			return
+		}
+		msg, err = nip44.Encrypt(string(payloadBytes), ck)
+		if err != nil {
+			logger.Logger.WithFields(logrus.Fields{
+				"notification": notification,
+				"appId":        app.ID,
+			}).WithError(err).Error("Failed to encrypt notification payload")
+			return
+		}
 	}
 
 	allTags := nostr.Tags{[]string{"p", app.AppPubkey}}
 	allTags = append(allTags, tags...)
 
-	appWalletPubKey, err := nostr.GetPublicKey(appWalletPrivKey)
-	if err != nil {
-		logger.Logger.WithFields(logrus.Fields{
-			"notification": notification,
-			"appId":        app.ID,
-		}).WithError(err).Error("Failed to calculate app wallet pub key")
-		return
-	}
-
 	event := &nostr.Event{
 		PubKey:    appWalletPubKey,
 		CreatedAt: nostr.Now(),
-		Kind:      models.NOTIFICATION_KIND,
 		Tags:      allTags,
 		Content:   msg,
 	}
+	if legacy {
+		event.Kind = models.LEGACY_NOTIFICATION_KIND
+	} else {
```

Comment:
rather than the `else` you could put the `Kind` back in the original declaration, and only overwrite it if we use `0.0`